rust/tw_bitcoin/src/modules/transactions/input_builder.rs

```diff
@@ -0,0 +1,320 @@
+use super::brc20::{BRC20TransferInscription, Brc20Ticker};
+use crate::aliases::*;
+use crate::modules::transactions::OrdinalNftInscription;
+use crate::{Error, Result};
+use bitcoin::taproot::{LeafVersion, TapLeafHash};
+use bitcoin::ScriptBuf;
+use secp256k1::XOnlyPublicKey;
+use tw_misc::traits::ToBytesVec;
+use tw_proto::BitcoinV2::Proto;
+use tw_proto::Utxo::Proto as UtxoProto;
+
+// Convenience varibles used solely for readability.
+const NO_LEAF_HASH: Option<TapLeafHash> = None;
+
+pub struct InputBuilder;
+
+impl InputBuilder {
+    pub fn utxo_from_proto(input: &Proto::Input<'_>) -> Result<UtxoProto::TxIn<'static>> {
```

Comment:
I believe this function can be useful in other Bitcoin forks like Decred, Verge etc

Reply:
Yes, this type is exposed after all and can be reused.